cpp/src/IceSSL/OpenSSLTransceiverI.cpp

```diff
@@ -339,6 +339,24 @@ OpenSSL::TransceiverI::initialize(IceInternal::Buffer& readBuffer, IceInternal::
     }
 
     _cipher = SSL_get_cipher_name(_ssl); // Nothing needs to be free'd.
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER < 0x10002000L
+    try
+    {
+        //
+        // Peer hostname verification is new in OpenSSL 1.0.2 for older versions
+        // we use IceSSL build in hostname verification.
+        //
+        _engine->verifyPeerCertName(address, info);
+    }
+    catch(cosnt SecurityException&)
```

Comment:
```suggestion
    catch(const SecurityException&)
```
We should make sure we still have a platform where this code is tested.

Reply:
Seems we are not testing OpenSSL < 1.0.2 with any of the supported platforms